### PR TITLE
Integrate wasmd 0.29.0-rc1

### PR DESCRIPTION
**Pull request by the-frey**
This version of wasmd includes: 

- support for patched IBC-go (3.3.0)
- different implementation of deterministic addresses from rc0
- support for SDK 45.8
- wasmvm 1.1.1 (mitigates cache error found on uni3)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,10 +20,10 @@ WORKDIR /code
 COPY . /code/
 
 # See https://github.com/CosmWasm/wasmvm/releases
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
-ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.0.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
-RUN sha256sum /lib/libwasmvm_muslc.aarch64.a | grep 7d2239e9f25e96d0d4daba982ce92367aacf0cbd95d2facb8442268f2b1cc1fc
-RUN sha256sum /lib/libwasmvm_muslc.x86_64.a | grep f6282df732a13dec836cda1f399dd874b1e3163504dbd9607c6af915b2740479
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.1.0/libwasmvm_muslc.aarch64.a /lib/libwasmvm_muslc.aarch64.a
+ADD https://github.com/CosmWasm/wasmvm/releases/download/v1.1.0/libwasmvm_muslc.x86_64.a /lib/libwasmvm_muslc.x86_64.a
```
Comment (webmaster128) [inline Dockerfile:24]: please use 1.1.1 to include the crash bug fix

Comment (webmaster128) [reply Dockerfile:24]: If you add a check like this, you make a mismatch between the runtime version and the source code version impossible: https://github.com/CosmWasm/wasmd/blob/v0.29.0-rc1/.circleci/config.yml#L167-L174

**Comment by the-frey** [comment]
This depends on #282

**Comment by giansalex** [comment]
new msg `/cosmwasm.wasm.v1.InstantiateContract2`  should be added to ICA host msgs

File: cmd/junod/genica.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+					"/cosmos.distribution.v1beta1.MsgSetWithdrawAddress",
+					"/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission",
+					"/cosmos.distribution.v1beta1.MsgFundCommunityPool",
+					"/cosmos.gov.v1beta1.MsgVote",
+					"/cosmos.authz.v1beta1.MsgExec",
+					"/cosmos.authz.v1beta1.MsgGrant",
+					"/cosmos.authz.v1beta1.MsgRevoke",
+					"/cosmwasm.wasm.v1.MsgStoreCode",
+					"/cosmwasm.wasm.v1.MsgInstantiateContract",
+					"/cosmwasm.wasm.v1.MsgExecuteContract",
```
Comment (giansalex) [inline cmd/junod/genica.go:63]: new msg `/cosmwasm.wasm.v1.InstantiateContract2`  should be added

**Comment by giansalex** [comment]
new msg `/cosmwasm.wasm.v1.InstantiateContract2`  should be added to ICA host msgs.

 I think `/cosmwasm.wasm.v1.MigrateContract` could be useful as well

**Pull request closed without merging**